pandas/tests/frame/test_mutate_columns.py

```diff
@@ -89,10 +89,26 @@ def test_assign_bad(self):
             df.assign(lambda x: x.A)
         with pytest.raises(AttributeError):
             df.assign(C=df.A, D=df.A + df.C)
-        with pytest.raises(KeyError):
-            df.assign(C=lambda df: df.A, D=lambda df: df['A'] + df['C'])
-        with pytest.raises(KeyError):
-            df.assign(C=df.A, D=lambda x: x['A'] + x['C'])
+        if not PY36:
```

Comment:
make this into a separate test (from the PY36) part and add a skipif PY36 decorator